### ISSUE TITLE
Select - fix colour contrast in secondary label

### ISSUE THREAD

**Issue by mkernohanbc** [bcgov/design-system#354]
The secondary label "(optional)" in the select component currently fails WCAG AA colour contrast requirement. The required contrast ratio is 4.5:1.

This label is currently styled using `--typography-color-disabled`. It should more properly use `--typography-color-secondary`, which should pass the contrast ratio requirement.

Axe test result in Storybook:
![Image](https://github.com/bcgov/design-system/assets/135075821/affd2f3c-e0cf-4d81-88d7-9badccb406c8)
